src/PostListingDisplay/components/PostListingDisplay.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { Component } from 'react';
+
+import InputField from './InputField';
+
+import '../styles/PostListingDisplay.css';
+
+/**Component class that displays the input fields for a user to post a listing */
+class PostListingDisplay extends Component {
+    constructor(props){
+        super(props);
+
+        this.state = {
+            title: '',
+            description: '',
+            numRooms: '',
+            numSingles: '',
+            singlePrice: '',
+            numShared: '',
+            sharedPrice: '',
+            numBathrooms: '',
+            leaseTypes: '',
+            startDate: '',
+            endDate: '',
+            listingPrice: ''
+        };
+
+        this.handleChange = this.handleChange.bind(this);
+        this.calculateListingPrice = this.calculateListingPrice.bind(this);
+    }
+
+    handleChange(event) {
+        const value = event.target.value;
+        const name = event.target.name;
+        this.setState({
+            [name]: value
+        })
+        this.calculateListingPrice();
```

Comment:
would it be possible to only calculate listing price if there's a change to the relevant fields (price or room type)?

Reply:
Done